### ISSUE TITLE
Ctrlr 5.6.23 (master branch) builds on Debian Bullseye, but crashes when opening a (new) panel

### ISSUE THREAD

**Issue by tomlem** [RomanKubiak/ctrlr#294]
After changing line 51 and line 93 of the Linux makefile into `VST3_PLATFORM_ARCH := $(shell uname -m)`  ctrlr 5.6.23 compiles on Debian Bullseye (g++ (Debian 10.2.1-6) 10.2.1 20210110). 
It opens, the preferences can be set and the build-in tools can be opened, but the midi device can not be set. When opening a panel (new or existing) it crashes (crashReport attached) while the output in the shell reads: `BFD::write_output  closed read handle: file: /home/tom/build/ctrlr/Builds/LinuxMakefile/build/Ctrlr temp:`.

Ctrlr.crash reads:

```
Ctrlr crash at: 17 May 2021 13:10:50
Stack trace:
./Ctrlr(+0x6767e7) [0x561592fbe7e7]
./Ctrlr(+0x5bb374) [0x561592f03374]
./Ctrlr(+0x665c2d) [0x561592fadc2d]
/lib/x86_64-linux-gnu/libpthread.so.0(+0x14140) [0x7f1e8c65b140]
./Ctrlr(+0x916b26) [0x56159325eb26]
./Ctrlr(+0x916ed4) [0x56159325eed4]
./Ctrlr(+0xa44c40) [0x56159338cc40]
./Ctrlr(+0xa4e42f) [0x56159339642f]
./Ctrlr(+0xa4ef36) [0x561593396f36]
./Ctrlr(+0x951478) [0x561593299478]
./Ctrlr(+0x93e842) [0x561593286842]
./Ctrlr(+0x5ca274) [0x561592f12274]
./Ctrlr(+0x6f6ec2) [0x56159303eec2]
./Ctrlr(+0x6f5e03) [0x56159303de03]
./Ctrlr(+0x6f5fd7) [0x56159303dfd7]
./Ctrlr(+0x2bcbbc) [0x561592c04bbc]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xea) [0x7f1e8bff3d0a]
./Ctrlr(+0x5bb0ba) [0x561592f030ba]
```

**Comment by keinstein** [comment]
Take a look at https://github.com/keinstein/ctrlr/tree/lua5.2 
Does it help?

**Comment by keinstein** [comment]
It could be a duplicate of #168 In that case the backtrace is useless.

**Comment by tomlem** [comment]
It doesn't build. Error code:
Makefile:336: *** target pattern contains no '%'. Stop.

**Comment by keinstein** [comment]
As far as I can see this is an issue with Projucer and VST3 plugins.
I always use one of
```
make -f pchbuild.mk CONFIG=Debug Standalone
make -f pchbuild.mk CONFIG=Release Standalone
```

**Comment by keinstein** [comment]
With the parameter `Standalone` the VST3 plugin does not get built, so the changes to the `Makefile` are not necessary.

Can you run Ctrlr from `gdb`?

```
gdb Ctrlr-Debug
r
bt full
```
and please, put the backtrace here?

you can quit gdb with the command `q` or `quit`.

There are several issues with JUCE assertions in Debug mode. We should take them seriously as they may lead to other errors and even crashes.

**Comment by tomlem** [comment]
ok

**Comment by tomlem** [comment]
~/build/ctrlr/Builds/LinuxMakefile/build$ gdb Ctrlr-Debug
GNU gdb (Debian 10.1-1.7) 10.1.90.20210103-git
Copyright (C) 2021 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<https://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from Ctrlr-Debug...
(gdb) r bt full
Starting program: /home/tom/build/ctrlr/Builds/LinuxMakefile/build/Ctrlr-Debug bt full
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
JUCE v6.0.7
CTRLR:initialise params "bt full"
[Detaching after fork from child process 5280]
[Detaching after fork from child process 5281]
[New Thread 0x7ffff5dcb700 (LWP 5284)]
CtrlrManagerVst::ctor
CtrlrLinux::getDefaultPanel, libr_open succeeded for binary [self]
BFD::write_output  closed read handle: file: /home/tom/build/ctrlr/Builds/LinuxMakefile/build/Ctrlr-Debug temp: 
[New Thread 0x7ffff54b2700 (LWP 5285)]
[Detaching after fork from child process 5287]
[Detaching after fork from child process 5288]
[Detaching after fork from child process 5289]
CtrlrLuaManager::ctor create debugger
[New Thread 0x7ffff4cb1700 (LWP 5319)]
[New Thread 0x7fffe7fff700 (LWP 5320)]
CtrlrPanel::restoreState
Open FAILED: [A]
Open FAILED: [B]
CtrlrPanel::bootstrapPanel
CtrlrPanel::setProgram
[New Thread 0x7fffe77fe700 (LWP 5321)]
Open OK: [E-MU XMidi1X1 Tab MIDI 1]
[New Thread 0x7fffe6ffd700 (LWP 5322)]
Open OK: [E-MU XMidi1X1 Tab MIDI 1]
CtrlrPanelWindowManager::windowClosedButtonPressed
[Detaching after fork from child process 5323]
[Detaching after fork from child process 5324]
JUCE Assertion failure in juce_String.cpp:324

Thread 1 "Ctrlr-Debug" received signal SIGTRAP, Trace/breakpoint trap.
0x00007ffff76df087 in kill () at ../sysdeps/unix/syscall-template.S:120
120	../sysdeps/unix/syscall-template.S: Bestand of map bestaat niet.
(gdb) q
A debugging session is active.

	Inferior 1 [process 5276] will be killed.

Quit anyway? (y or n) y

**Comment by tomlem** [comment]
~/build/ctrlr/Builds/LinuxMakefile/build$ gdb Ctrlr-Debug
r
bt full
GNU gdb (Debian 10.1-1.7) 10.1.90.20210103-git
Copyright (C) 2021 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<https://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from Ctrlr-Debug...
(gdb) r
Starting program: /home/tom/build/ctrlr/Builds/LinuxMakefile/build/Ctrlr-Debug 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
JUCE v6.0.7
CTRLR:initialise params ""
[Detaching after fork from child process 5035]
[Detaching after fork from child process 5036]
[New Thread 0x7ffff5dcb700 (LWP 5039)]
CtrlrManagerVst::ctor
CtrlrLinux::getDefaultPanel, libr_open succeeded for binary [self]
BFD::write_output  closed read handle: file: /home/tom/build/ctrlr/Builds/LinuxMakefile/build/Ctrlr-Debug temp: 
CtrlrManager::restoreState enter
CtrlrManager::restoreState exit
[New Thread 0x7ffff54b2700 (LWP 5040)]
bt full
[Detaching after fork from child process 5043]
[Detaching after fork from child process 5044]
[Detaching after fork from child process 5045]
CtrlrStandaloneWindow::saveStateNow
CtrlrProcessor::getStateInformation
[Thread 0x7ffff5dcb700 (LWP 5039) exited]
CtrlrManagerVst::dtor
[Thread 0x7ffff54b2700 (LWP 5040) exited]
[Inferior 1 (process 5031) exited normally]
(gdb) bt full
No stack.
(gdb) 
No stack.
(gdb) q
bash: r: opdracht niet gevonden
bash: bt: opdracht niet gevonden
tom@paradigit:~/build/ctrlr/Builds/LinuxMakefile/build$ gdb Ctrlr-Debug
GNU gdb (Debian 10.1-1.7) 10.1.90.20210103-git
Copyright (C) 2021 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<https://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from Ctrlr-Debug...
(gdb) r bt full
Starting program: /home/tom/build/ctrlr/Builds/LinuxMakefile/build/Ctrlr-Debug bt full
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
JUCE v6.0.7
CTRLR:initialise params "bt full"
[Detaching after fork from child process 5094]
[Detaching after fork from child process 5095]
[New Thread 0x7ffff5dcb700 (LWP 5098)]
CtrlrManagerVst::ctor
CtrlrLinux::getDefaultPanel, libr_open succeeded for binary [self]
BFD::write_output  closed read handle: file: /home/tom/build/ctrlr/Builds/LinuxMakefile/build/Ctrlr-Debug temp: 
CtrlrManager::restoreState enter
CtrlrManager::restoreState exit
[New Thread 0x7ffff54b2700 (LWP 5099)]
[Detaching after fork from child process 5100]
[Detaching after fork from child process 5101]
[Detaching after fork from child process 5102]
CtrlrLuaManager::ctor create debugger
[New Thread 0x7ffff4cb1700 (LWP 5137)]
[New Thread 0x7fffeffff700 (LWP 5138)]
CtrlrPanel::restoreState
Open FAILED: [A]
Open FAILED: [B]
CtrlrPanel::bootstrapPanel
CtrlrPanel::setProgram
[Detaching after fork from child process 5139]
[Detaching after fork from child process 5140]
JUCE Assertion failure in juce_String.cpp:324

Thread 1 "Ctrlr-Debug" received signal SIGTRAP, Trace/breakpoint trap.
0x00007ffff76df087 in kill () at ../sysdeps/unix/syscall-template.S:120
120	../sysdeps/unix/syscall-template.S: Bestand of map bestaat niet.
(gdb) q
A debugging session is active.

	Inferior 1 [process 5090] will be killed.

Quit anyway? (y or n) y
tom@paradigit:~/build/ctrlr/Builds/LinuxMakefile/build$ gdb Ctrlr-Debug
GNU gdb (Debian 10.1-1.7) 10.1.90.20210103-git
Copyright (C) 2021 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<https://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from Ctrlr-Debug...
(gdb) r bt full
Starting program: /home/tom/build/ctrlr/Builds/LinuxMakefile/build/Ctrlr-Debug bt full
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
JUCE v6.0.7
CTRLR:initialise params "bt full"
[Detaching after fork from child process 5181]
[Detaching after fork from child process 5182]
[New Thread 0x7ffff5dcb700 (LWP 5185)]
CtrlrManagerVst::ctor
CtrlrLinux::getDefaultPanel, libr_open succeeded for binary [self]
BFD::write_output  closed read handle: file: /home/tom/build/ctrlr/Builds/LinuxMakefile/build/Ctrlr-Debug temp: 
CtrlrManager::restoreState enter
CtrlrManager::restoreState exit
[New Thread 0x7ffff54b2700 (LWP 5187)]
[Detaching after fork from child process 5188]
[Detaching after fork from child process 5189]
[Detaching after fork from child process 5190]
CtrlrLuaManager::ctor create debugger
[New Thread 0x7ffff4cb1700 (LWP 5216)]
[New Thread 0x7fffeffff700 (LWP 5217)]
CtrlrPanel::restoreState
Open FAILED: [A]
Open FAILED: [B]
CtrlrPanel::bootstrapPanel
CtrlrPanel::setProgram
[New Thread 0x7fffef7fe700 (LWP 5221)]
Open OK: [E-MU XMidi1X1 Tab MIDI 1]
[New Thread 0x7fffeeffd700 (LWP 5222)]
Open OK: [E-MU XMidi1X1 Tab MIDI 1]
CtrlrPanelWindowManager::windowClosedButtonPressed
[Detaching after fork from child process 5223]
[Detaching after fork from child process 5224]
JUCE Assertion failure in juce_String.cpp:324

Thread 1 "Ctrlr-Debug" received signal SIGTRAP, Trace/breakpoint trap.
0x00007ffff76df087 in kill () at ../sysdeps/unix/syscall-template.S:120
120	../sysdeps/unix/syscall-template.S: Bestand of map bestaat niet.
(gdb) q
A debugging session is active.

	Inferior 1 [process 5177] will be killed.

Quit anyway? (y or n) y
tom@paradigit:~/build/ctrlr/Builds/LinuxMakefile/build$ gdb Ctrlr-Debug
GNU gdb (Debian 10.1-1.7) 10.1.90.20210103-git
Copyright (C) 2021 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<https://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from Ctrlr-Debug...
(gdb) r bt full
Starting program: /home/tom/build/ctrlr/Builds/LinuxMakefile/build/Ctrlr-Debug bt full
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
JUCE v6.0.7
CTRLR:initialise params "bt full"
[Detaching after fork from child process 5280]
[Detaching after fork from child process 5281]
[New Thread 0x7ffff5dcb700 (LWP 5284)]
CtrlrManagerVst::ctor
CtrlrLinux::getDefaultPanel, libr_open succeeded for binary [self]
BFD::write_output  closed read handle: file: /home/tom/build/ctrlr/Builds/LinuxMakefile/build/Ctrlr-Debug temp: 
[New Thread 0x7ffff54b2700 (LWP 5285)]
[Detaching after fork from child process 5287]
[Detaching after fork from child process 5288]
[Detaching after fork from child process 5289]
CtrlrLuaManager::ctor create debugger
[New Thread 0x7ffff4cb1700 (LWP 5319)]
[New Thread 0x7fffe7fff700 (LWP 5320)]
CtrlrPanel::restoreState
Open FAILED: [A]
Open FAILED: [B]
CtrlrPanel::bootstrapPanel
CtrlrPanel::setProgram
[New Thread 0x7fffe77fe700 (LWP 5321)]
Open OK: [E-MU XMidi1X1 Tab MIDI 1]
[New Thread 0x7fffe6ffd700 (LWP 5322)]
Open OK: [E-MU XMidi1X1 Tab MIDI 1]
CtrlrPanelWindowManager::windowClosedButtonPressed
[Detaching after fork from child process 5323]
[Detaching after fork from child process 5324]
JUCE Assertion failure in juce_String.cpp:324

Thread 1 "Ctrlr-Debug" received signal SIGTRAP, Trace/breakpoint trap.
0x00007ffff76df087 in kill () at ../sysdeps/unix/syscall-template.S:120
120	../sysdeps/unix/syscall-template.S: Bestand of map bestaat niet.
(gdb) q
A debugging session is active.

	Inferior 1 [process 5276] will be killed.

Quit anyway? (y or n) y
tom@paradigit:~/build/ctrlr/Builds/LinuxMakefile/build$ gdb Ctrlr-Debug
GNU gdb (Debian 10.1-1.7) 10.1.90.20210103-git
Copyright (C) 2021 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<https://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from Ctrlr-Debug...
(gdb) r
Starting program: /home/tom/build/ctrlr/Builds/LinuxMakefile/build/Ctrlr-Debug 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
JUCE v6.0.7
CTRLR:initialise params ""
[Detaching after fork from child process 5436]
[Detaching after fork from child process 5437]
[New Thread 0x7ffff5dcb700 (LWP 5440)]
CtrlrManagerVst::ctor
CtrlrLinux::getDefaultPanel, libr_open succeeded for binary [self]
BFD::write_output  closed read handle: file: /home/tom/build/ctrlr/Builds/LinuxMakefile/build/Ctrlr-Debug temp: 
[New Thread 0x7ffff54b2700 (LWP 5441)]
[Detaching after fork from child process 5443]
[Detaching after fork from child process 5444]
[Detaching after fork from child process 5445]
CtrlrLuaManager::ctor create debugger
[New Thread 0x7ffff4cb1700 (LWP 5477)]
[New Thread 0x7fffeffff700 (LWP 5478)]
CtrlrPanel::restoreState
Open FAILED: [A]
Open FAILED: [B]
CtrlrPanel::bootstrapPanel
CtrlrPanel::setProgram
[New Thread 0x7fffef7fe700 (LWP 5481)]
Open OK: [E-MU XMidi1X1 Tab MIDI 1]
[New Thread 0x7fffeeffd700 (LWP 5482)]
Open OK: [E-MU XMidi1X1 Tab MIDI 1]
CtrlrPanelWindowManager::windowClosedButtonPressed
[Detaching after fork from child process 5483]
[Detaching after fork from child process 5484]
JUCE Assertion failure in juce_String.cpp:324

Thread 1 "Ctrlr-Debug" received signal SIGTRAP, Trace/breakpoint trap.
0x00007ffff76df087 in kill () at ../sysdeps/unix/syscall-template.S:120
120	../sysdeps/unix/syscall-template.S: Bestand of map bestaat niet.
(gdb) bt full
#0  0x00007ffff76df087 in kill () at ../sysdeps/unix/syscall-template.S:120
No locals.
#1  0x0000555555964381 in juce::String::String (this=0x7fffffffcdb8, 
    t=0x555557904498 "Loaded \273 180 bytes from sysex file")
    at ../../JUCE/modules/juce_core/text/juce_String.cpp:324
No locals.
#2  0x000055555573275c in luabind::default_converter<juce::String, void>::from
    (this=0x7fffffffcda7, L=0x55555799e500, index=2)
    at ../../Source/Core/stdafx_luabind.h:162
        obj = {<luabind::adl::object_interface<luabind::adl::object>> = {<No data fields>}, m_handle = {m_interpreter = 0x55555799e500, m_index = 835}}
        name = "string"
#3  0x000055555574fabf in luabind::native_converter_base<juce::String, luabind::default_converter<juce::String, void> >::apply (this=0x7fffffffcda7, 
    L=0x55555799e500, index=2)
    at ../../Source/Misc/luabind/luabind/detail/policy.hpp:662
No locals.
#4  0x00005555561e70f8 in luabind::detail::invoke_member<CtrlrLabel& (CtrlrLabel::*)(juce::String const&), boost::function_types::components<CtrlrLabel& (CtrlrLabel::*)(juce::String const&), boost::add_reference<luabind::detail::most_derived<mpl_::arg<1>, CtrlrLabel> > >, luabind::detail::null_type> (
    L=0x55555799e500, self=..., ctx=..., f=
    @0x555557bcdc68: (CtrlrLabel &(CtrlrLabel::*)(CtrlrLabel * const, const juce--Type <RET> for more, q to quit, c to continue without paging--
::String &)) 0x555556114922 <CtrlrLabel::setLabelText(juce::String const&)>)
    at ../../Source/Misc/luabind/luabind/detail/call.hpp:298
        indices = {21845, -1, -1}
        result_converter = {<luabind::detail::ref_converter> = {<luabind::detail::pointer_converter> = {result = 0x0}, <No data fields>}, <No data fields>}
        c0 = {<luabind::detail::ref_converter> = {<luabind::detail::pointer_converter> = {result = 0x555557ccd630}, <No data fields>}, <No data fields>}
        index0 = 1
        c1 = {<luabind::default_converter<juce::String, void>> = {<luabind::native_converter_base<juce::String, luabind::default_converter<juce::String, void> >> = {<No data fields>}, <No data fields>}, <No data fields>}
        index1 = 2
        arity = 2
        arguments = 2
        score = 0
        results = 0
#5  0x00005555561e43c2 in luabind::detail::invoke0<CtrlrLabel& (CtrlrLabel::*)(juce::String const&), boost::function_types::components<CtrlrLabel& (CtrlrLabel::*)(juce::String const&), boost::add_reference<luabind::detail::most_derived<mpl_::arg<1>, CtrlrLabel> > >, luabind::detail::null_type, boost::is_void<CtrlrLabel&> > (L=0x55555799e500, self=..., ctx=..., f=
    @0x555557bcdc68: (CtrlrLabel &(CtrlrLabel::*)(CtrlrLabel * const, const juce::String &)) 0x555556114922 <CtrlrLabel::setLabelText(juce::String const&)>, 
--Type <RET> for more, q to quit, c to continue without paging--
    policies=...) at ../../Source/Misc/luabind/luabind/detail/call.hpp:80
No locals.
#6  0x00005555561e371b in luabind::detail::invoke<CtrlrLabel& (CtrlrLabel::*)(juce::String const&), boost::function_types::components<CtrlrLabel& (CtrlrLabel::*)(juce::String const&), boost::add_reference<luabind::detail::most_derived<mpl_::arg<1>, CtrlrLabel> > >, luabind::detail::null_type> (L=0x55555799e500, 
    self=..., ctx=..., f=
    @0x555557bcdc68: (CtrlrLabel &(CtrlrLabel::*)(CtrlrLabel * const, const juce::String &)) 0x555556114922 <CtrlrLabel::setLabelText(juce::String const&)>, 
    policies=...) at ../../Source/Misc/luabind/luabind/detail/call.hpp:102
No locals.
#7  0x00005555561e0c75 in luabind::detail::function_object_impl<CtrlrLabel& (CtrlrLabel::*)(juce::String const&), boost::function_types::components<CtrlrLabel& (CtrlrLabel::*)(juce::String const&), boost::add_reference<luabind::detail::most_derived<mpl_::arg<1>, CtrlrLabel> > >, luabind::detail::null_type>::entry_point (L=0x55555799e500) at ../../Source/Misc/luabind/luabind/make_function.hpp:62
        ctx = {static max_candidates = 10, candidates = {0x555557bcdc20, 
            0x555555d21009 <newlstr+181>, 0x4fd5799e500, 0x22, 0x555557ced420, 
            0x55555799e500, 0x55555799e5b8, 0x555557904480, 0x7fffffffcf50, 
            0x7ffff77a4164 <mcount+52>}, best_score = 0, candidate_index = 1, 
          additional_candidates = 0}
        __PRETTY_FUNCTION__ = "static int luabind::detail::function_object_impl<F, Signature, Policies>::entry_point(lua_State*) [with F = CtrlrLabel& (CtrlrLab--Type <RET> for more, q to quit, c to continue without paging--
el::*)(const juce::String&); Signature = boost::function_types::compone"...
        impl = 0x555557bcdc20
        results = 0
        error = false
#8  0x0000555555d16868 in luaD_precall (L=0x55555799e500, func=0x555557c0cfa0, 
    nresults=0) at ../../Source/Misc/lua/src/lua.c:5465
        ci = 0x55555799e7e8
        n = 21845
        cl = 0x555557bcdcd0
        funcr = 128
#9  0x0000555555d32ac9 in luaV_execute (L=0x55555799e500, nexeccalls=2)
    at ../../Source/Misc/lua/src/lua.c:15420
        b = 3
        nresults = 0
        i = 25182364
        ra = 0x555557c0cfa0
        cl = 0x555557c49cc0
        base = 0x555557c0cf80
        k = 0x555557c4b060
        pc = 0x555557c4b390
#10 0x0000555555d16b1d in luaD_call (L=0x55555799e500, func=0x555557c0cf40, 
    nResults=0) at ../../Source/Misc/lua/src/lua.c:5523
No locals.
--Type <RET> for more, q to quit, c to continue without paging--
#11 0x0000555555d10aac in f_call (L=0x55555799e500, ud=0x7fffffffd5a0)
    at ../../Source/Misc/lua/src/lua.c:2649
        c = 0x7fffffffd5a0
#12 0x0000555555d15b9b in luaD_rawrunprotected (L=0x55555799e500, 
    f=0x555555d10a71 <f_call>, ud=0x7fffffffd5a0)
    at ../../Source/Misc/lua/src/lua.c:5262
        lj = {previous = 0x0, b = {{__jmpbuf = {93825028961232, 
                -5315823126270956922, 41, 0, 16, 0, -5315823126237402490, 
                -2058370039133492602}, __mask_was_saved = 0, __saved_mask = {
                __val = {140737488344208, 93824994008710, 93825023634984, 
                  93825024083368, 140737488344272, 140737345372516, 0, 
                  93825028961232, 140737488344320, 41, 93825028961232, 
                  140737345372516, 93825030284544, 32, 140737488344480, 
                  93825000344177}}}}, status = 0}
#13 0x0000555555d16f2f in luaD_pcall (L=0x55555799e500, 
    func=0x555555d10a71 <f_call>, u=0x7fffffffd5a0, old_top=32, ef=16)
    at ../../Source/Misc/lua/src/lua.c:5609
        status = 21845
        oldnCcalls = 0
        old_ci = 0
        old_allowhooks = 1 '\001'
        old_errfunc = 0
#14 0x0000555555d10b5d in lua_pcall (L=0x55555799e500, nargs=2, nresults=0, 
--Type <RET> for more, q to quit, c to continue without paging--
    errfunc=1) at ../../Source/Misc/lua/src/lua.c:2670
        c = {func = 0x555557c0cf40, nresults = 0}
        status = 21845
        func = 16
#15 0x00005555563cd87e in luabind::detail::pcall (L=0x55555799e500, nargs=2, 
    nresults=0) at ../../Source/Misc/luabind/src/pcall.cpp:42
        e = 0x555555728017 <add_file_and_line(lua_State*)>
        en = 1
        result = 1442824998
#16 0x0000555555ff6b9b in luabind::detail::proxy_function_void_caller<boost::tuples::tuple<CtrlrComponent* const*, juce::MouseEvent const*, boost::tuples::null_type, boost::tuples::null_type, boost::tuples::null_type, boost::tuples::null_type, boost::tuples::null_type, boost::tuples::null_type, boost::tuples::null_type, boost::tuples::null_type> >::~proxy_function_void_caller (
    this=0x7fffffffd6b0, __in_chrg=<optimized out>)
    at ../../Source/Misc/luabind/luabind/detail/call_function.hpp:255
        L = 0x55555799e500
        top = 1
        pop = {m_state = 0x7fffffffd678, m_n = -9808}
        L = <optimized out>
        top = <optimized out>
        pop = {m_state = <optimized out>, m_n = <optimized out>}
        __PRETTY_FUNCTION__ = {<optimized out> <repeats 375 times>}
--Type <RET> for more, q to quit, c to continue without paging--
        L = <optimized out>
        top = <optimized out>
        pop = {m_state = <optimized out>, m_n = <optimized out>}
#17 0x0000555555fdb98a in CtrlrLuaMethodManager::call (this=0x55555785b3d0, o=
    0x555557c46590, param1=0x55555784ef30, param2=...)
    at ../../Source/Lua/Methods/CtrlrLuaMethodManagerCalls.cpp:187
        sl = {lock_ = @0x55555785b4f0}
#18 0x00005555561a9314 in CtrlrComponent::mouseDown (this=0x55555784ef30, 
    e=...) at ../../Source/UIComponents/CtrlrComponents/CtrlrComponent.cpp:193
No locals.
#19 0x00005555560d7987 in CtrlrButton::mouseDown (this=0x55555784ef30, e=...)
    at ../../Source/UIComponents/CtrlrComponents/Buttons/CtrlrButton.cpp:114
No locals.
#20 0x0000555555c16e90 in juce::Component::MouseListenerList::sendMouseEvent<juce::MouseEvent const&> (comp=..., checker=..., 
    eventMethod=&virtual juce::MouseListener::mouseDown(juce::MouseEvent const&)) at ../../JUCE/modules/juce_gui_basics/components/juce_Component.cpp:111
        i = 0
        list = 0x5555577bc0d0
#21 0x0000555555b1599a in juce::Component::internalMouseDown (
    this=0x555557cb5560, source=..., relativePos=..., time=..., pressure=0, 
    orientation=0, rotation=0, tiltX=0, tiltY=0)
    at ../../JUCE/modules/juce_gui_basics/components/juce_Component.cpp:2432
--Type <RET> for more, q to quit, c to continue without paging--
        desktop = @0x555557733220: {<juce::DeletedAtShutdown> = {
            _vptr.DeletedAtShutdown = 0x555557321398 <vtable for juce::Desktop+16>}, <juce::Timer> = {
            _vptr.Timer = 0x5555573213c8 <vtable for juce::Desktop+64>, 
            positionInQueue = 4, timerPeriodMs = 0}, <juce::AsyncUpdater> = {
            _vptr.AsyncUpdater = 0x5555573213f0 <vtable for juce::Desktop+104>, activeMessage = {referencedObject = 0x555557732fc0}, 
            leakDetector107 = {<No data fields>}}, 
          static instance = 0x555557733220, 
          mouseSources = std::unique_ptr<juce::MouseInputSource::SourceList> = {get() = 0x5555577333d0}, mouseListeners = {listeners = {
              values = {<juce::DummyCriticalSection> = {<No data fields>}, 
                elements = {data = 0x555557908640}, numAllocated = 8, 
                numUsed = 0}}}, focusListeners = {listeners = {
              values = {<juce::DummyCriticalSection> = {<No data fields>}, 
                elements = {data = 0x555557752480}, numAllocated = 8, 
                numUsed = 2}}}, desktopComponents = {
            values = {<juce::DummyCriticalSection> = {<No data fields>}, 
              elements = {data = 0x55555774c0a0}, numAllocated = 8, 
              numUsed = 2}}, peers = {
            values = {<juce::DummyCriticalSection> = {<No data fields>}, 
              elements = {data = 0x555557738cc0}, numAllocated = 8, 
              numUsed = 2}}, displays = std::unique_ptr<juce::Displays> = {
--Type <RET> for more, q to quit, c to continue without paging--
            get() = 0x5555577338f0}, lastFakeMouseMove = {x = 813, y = 1034}, 
          mouseClickCounter = 13, mouseWheelCounter = 0, 
          defaultLookAndFeel = std::unique_ptr<juce::LookAndFeel> = {
            get() = 0x55555774c700}, currentLookAndFeel = {holder = {
              referencedObject = 0x5555577d64e0}}, kioskModeComponent = 0x0, 
          kioskComponentOriginalBounds = {pos = {x = 0, y = 0}, w = 0, h = 0}, 
          kioskModeReentrant = false, allowedOrientations = 15, 
          masterScaleFactor = 1, animator = {<juce::ChangeBroadcaster> = {
              _vptr.ChangeBroadcaster = 0x55555731d540 <vtable for juce::ComponentAnimator+16>, broadcastCallback = {<juce::AsyncUpdater> = {
                  _vptr.AsyncUpdater = 0x55555730d860 <vtable for juce::ChangeBroadcaster::ChangeBroadcasterCallback+16>, activeMessage = {
                    referencedObject = 0x5555577338c0}, 
                  leakDetector107 = {<No data fields>}}, 
                owner = 0x5555577332f0}, changeListeners = {listeners = {
                  values = {<juce::DummyCriticalSection> = {<No data fields>}, 
                    elements = {data = 0x0}, numAllocated = 0, numUsed = 0}}}, 
              anyListeners = {_M_base = {static _S_alignment = 1, 
                  _M_i = false}}}, <juce::Timer> = {
              _vptr.Timer = 0x55555731d568 <vtable for juce::ComponentAnimator+56>, positionInQueue = 18446744073709551615, timerPeriodMs = 0}, tasks = {
              values = {<juce::DummyCriticalSection> = {<No data fields>}, 
                elements = {data = 0x0}, numAllocated = 0, numUsed = 0}, 
--Type <RET> for more, q to quit, c to continue without paging--
              leakDetector870 = {<No data fields>}}, lastTime = 0, 
            leakDetector159 = {<No data fields>}}, 
          leakDetector420 = {<No data fields>}}
        checker = {safePointer = {holder = {
              referencedObject = 0x5555577bc490}}}
        me = {position = {x = 64, y = 19}, x = 64, y = 19, mods = {
            static currentModifiers = {
              static currentModifiers = <same as static member of an already seen type>, flags = 0}, flags = 16}, pressure = 0, orientation = 0, rotation = 0, 
          tiltX = 0, tiltY = 0, mouseDownPosition = {x = 64, y = 19}, 
          eventComponent = 0x555557cb5560, originalComponent = 0x555557cb5560, 
          eventTime = {millisSinceEpoch = 1624286158196}, mouseDownTime = {
            millisSinceEpoch = 1624286158196}, source = {
            static invalidPressure = 0, static invalidOrientation = 0, 
            static invalidRotation = 0, static invalidTiltX = 0, 
            static invalidTiltY = 0, static offscreenMousePos = {x = -10, 
              y = -10}, pimpl = 0x55555746c260, 
            leakDetector268 = {<No data fields>}}, numberOfClicks = 1 '\001', 
          wasMovedSinceMouseDown = 0 '\000'}
#22 0x0000555555bd187e in juce::MouseInputSourceInternal::sendMouseDown (
    this=0x55555746c260, comp=..., screenPos=..., time=...)
    at ../../JUCE/modules/juce_gui_basics/mouse/juce_MouseInputSource.cpp:142
No locals.
--Type <RET> for more, q to quit, c to continue without paging--
#23 0x0000555555bd1dfb in juce::MouseInputSourceInternal::setButtons (
    this=0x55555746c260, screenPos=..., time=..., newButtonState=...)
    at ../../JUCE/modules/juce_gui_basics/mouse/juce_MouseInputSource.cpp:215
        current = 0x555557cb5560
        lastCounter = 1018
#24 0x0000555555bd244f in juce::MouseInputSourceInternal::handleEvent (
    this=0x55555746c260, newPeer=..., positionWithinPeer=..., time=..., 
    newMods=..., newPressure=0, newOrientation=0, pen=...)
    at ../../JUCE/modules/juce_gui_basics/mouse/juce_MouseInputSource.cpp:333
        peer = 0x55555774bcb0
        pressureChanged = false
        orientationChanged = false
        rotationChanged = false
        tiltChanged = false
        shouldUpdate = false
        screenPos = {x = 758, y = 1052}
#25 0x0000555555b18997 in juce::MouseInputSource::handleEvent (
    this=0x7fffffffdc90, peer=..., pos=..., time=1624286158196, mods=..., 
    pressure=0, orientation=0, penDetails=...)
    at ../../JUCE/modules/juce_gui_basics/mouse/juce_MouseInputSource.cpp:633
No locals.
#26 0x0000555555b9d926 in juce::ComponentPeer::handleMouseEvent (
    this=0x55555774bcb0, type=juce::MouseInputSource::mouse, pos=..., 
--Type <RET> for more, q to quit, c to continue without paging--
    newMods=..., newPressure=0, newOrientation=0, time=1624286158196, pen=..., 
    touchIndex=0)
    at ../../JUCE/modules/juce_gui_basics/windows/juce_ComponentPeer.cpp:87
        mouse = 0x55555746c050
#27 0x0000555555bb9114 in juce::XWindowSystem::handleButtonPressEvent (
    this=0x555557733950, peer=0x55555774bcb0, buttonPressEvent=..., 
    buttonModifierFlag=16)
    at ../../JUCE/modules/juce_gui_basics/native/x11/juce_linux_XWindowSystem.cpp:3161
No locals.
#28 0x0000555555bb9208 in juce::XWindowSystem::handleButtonPressEvent (
    this=0x555557733950, peer=0x55555774bcb0, buttonPressEvent=...)
    at ../../JUCE/modules/juce_gui_basics/native/x11/juce_linux_XWindowSystem.cpp:3178
        mapIndex = 0
#29 0x0000555555bb862b in juce::XWindowSystem::handleWindowMessage (
    this=0x555557733950, peer=0x55555774bcb0, event=...)
    at ../../JUCE/modules/juce_gui_basics/native/x11/juce_linux_XWindowSystem.cpp:2944
No locals.
#30 0x0000555555bba35f in juce::WindowingHelpers::windowMessageReceive (
    event=...)
    at ../../JUCE/modules/juce_gui_basics/native/x11/juce_linux_XWindowSystem.cp--Type <RET> for more, q to quit, c to continue without paging--
p:3465
        peer = 0x55555774bcb0
#31 0x0000555555bb7ed4 in operator() (__closure=0x555557732ca0)
    at ../../JUCE/modules/juce_gui_basics/native/x11/juce_linux_XWindowSystem.cpp:2874
        evt = {type = 4, xany = {type = 4, serial = 12816, send_event = 0, 
            display = 0x555557736720, window = 35651594}, xkey = {type = 4, 
            serial = 12816, send_event = 0, display = 0x555557736720, 
            window = 35651594, root = 1314, subwindow = 0, time = 5126592, 
            x = 295, y = 399, x_root = 758, y_root = 1052, state = 16, 
            keycode = 1, same_screen = 1}, xbutton = {type = 4, 
            serial = 12816, send_event = 0, display = 0x555557736720, 
            window = 35651594, root = 1314, subwindow = 0, time = 5126592, 
            x = 295, y = 399, x_root = 758, y_root = 1052, state = 16, 
            button = 1, same_screen = 1}, xmotion = {type = 4, serial = 12816, 
            send_event = 0, display = 0x555557736720, window = 35651594, 
            root = 1314, subwindow = 0, time = 5126592, x = 295, y = 399, 
            x_root = 758, y_root = 1052, state = 16, is_hint = 1 '\001', 
            same_screen = 1}, xcrossing = {type = 4, serial = 12816, 
            send_event = 0, display = 0x555557736720, window = 35651594, 
            root = 1314, subwindow = 0, time = 5126592, x = 295, y = 399, 
            x_root = 758, y_root = 1052, mode = 16, detail = 1, 
            same_screen = 1, focus = -1, state = 0}, xfocus = {type = 4, 
--Type <RET> for more, q to quit, c to continue without paging--
            serial = 12816, send_event = 0, display = 0x555557736720, 
            window = 35651594, mode = 1314, detail = 0}, xexpose = {type = 4, 
            serial = 12816, send_event = 0, display = 0x555557736720, 
            window = 35651594, x = 1314, y = 0, width = 0, height = 0, 
            count = 5126592}, xgraphicsexpose = {type = 4, serial = 12816, 
            send_event = 0, display = 0x555557736720, drawable = 35651594, 
            x = 1314, y = 0, width = 0, height = 0, count = 5126592, 
            major_code = 0, minor_code = 295}, xnoexpose = {type = 4, 
            serial = 12816, send_event = 0, display = 0x555557736720, 
            drawable = 35651594, major_code = 1314, minor_code = 0}, 
          xvisibility = {type = 4, serial = 12816, send_event = 0, 
            display = 0x555557736720, window = 35651594, state = 1314}, 
          xcreatewindow = {type = 4, serial = 12816, send_event = 0, 
            display = 0x555557736720, parent = 35651594, window = 1314, x = 0, 
            y = 0, width = 5126592, height = 0, border_width = 295, 
            override_redirect = 399}, xdestroywindow = {type = 4, 
            serial = 12816, send_event = 0, display = 0x555557736720, 
            event = 35651594, window = 1314}, xunmap = {type = 4, 
            serial = 12816, send_event = 0, display = 0x555557736720, 
            event = 35651594, window = 1314, from_configure = 0}, xmap = {
            type = 4, serial = 12816, send_event = 0, 
            display = 0x555557736720, event = 35651594, window = 1314, 
            override_redirect = 0}, xmaprequest = {type = 4, serial = 12816, 
--Type <RET> for more, q to quit, c to continue without paging--
            send_event = 0, display = 0x555557736720, parent = 35651594, 
            window = 1314}, xreparent = {type = 4, serial = 12816, 
            send_event = 0, display = 0x555557736720, event = 35651594, 
            window = 1314, parent = 0, x = 5126592, y = 0, 
            override_redirect = 295}, xconfigure = {type = 4, serial = 12816, 
            send_event = 0, display = 0x555557736720, event = 35651594, 
            window = 1314, x = 0, y = 0, width = 5126592, height = 0, 
            border_width = 295, above = 4518305596150, 
            override_redirect = 16}, xgravity = {type = 4, serial = 12816, 
            send_event = 0, display = 0x555557736720, event = 35651594, 
            window = 1314, x = 0, y = 0}, xresizerequest = {type = 4, 
            serial = 12816, send_event = 0, display = 0x555557736720, 
            window = 35651594, width = 1314, height = 0}, xconfigurerequest = {
            type = 4, serial = 12816, send_event = 0, 
            display = 0x555557736720, parent = 35651594, window = 1314, x = 0, 
            y = 0, width = 5126592, height = 0, border_width = 295, 
            above = 4518305596150, detail = 16, 
            value_mask = 18446744069414584321}, xcirculate = {type = 4, 
            serial = 12816, send_event = 0, display = 0x555557736720, 
            event = 35651594, window = 1314, place = 0}, xcirculaterequest = {
            type = 4, serial = 12816, send_event = 0, 
            display = 0x555557736720, parent = 35651594, window = 1314, 
            place = 0}, xproperty = {type = 4, serial = 12816, send_event = 0, 
--Type <RET> for more, q to quit, c to continue without paging--
            display = 0x555557736720, window = 35651594, atom = 1314, 
            time = 0, state = 5126592}, xselectionclear = {type = 4, 
            serial = 12816, send_event = 0, display = 0x555557736720, 
            window = 35651594, selection = 1314, time = 0}, 
          xselectionrequest = {type = 4, serial = 12816, send_event = 0, 
            display = 0x555557736720, owner = 35651594, requestor = 1314, 
            selection = 0, target = 5126592, property = 1713691951399, 
            time = 4518305596150}, xselection = {type = 4, serial = 12816, 
            send_event = 0, display = 0x555557736720, requestor = 35651594, 
            selection = 1314, target = 0, property = 5126592, 
            time = 1713691951399}, xcolormap = {type = 4, serial = 12816, 
            send_event = 0, display = 0x555557736720, window = 35651594, 
            colormap = 1314, c_new = 0, state = 0}, xclient = {type = 4, 
            serial = 12816, send_event = 0, display = 0x555557736720, 
            window = 35651594, message_type = 1314, format = 0, data = {
              b = "\300\071N\000\000\000\000\000'\001\000\000\217\001\000\000\366\002\000", s = {14784, 78, 0, 0, 295, 0, 399, 0, 758, 0}, l = {5126592, 
                1713691951399, 4518305596150, 4294967312, -4294967295}}}, 
          xmapping = {type = 4, serial = 12816, send_event = 0, 
            display = 0x555557736720, window = 35651594, request = 1314, 
            first_keycode = 0, count = 0}, xerror = {type = 4, 
            display = 0x3210, resourceid = 0, serial = 93825027761952, 
            error_code = 10 '\n', request_code = 0 '\000', 
--Type <RET> for more, q to quit, c to continue without paging--
            minor_code = 32 ' '}, xkeymap = {type = 4, serial = 12816, 
            send_event = 0, display = 0x555557736720, window = 35651594, 
            key_vector = "\"\005", '\000' <repeats 14 times>, "\300\071N\000\000\000\000\000'\001\000\000\217\001\000"}, xgeneric = {type = 4, serial = 12816, 
            send_event = 0, display = 0x555557736720, extension = 35651594, 
            evtype = 0}, xcookie = {type = 4, serial = 12816, send_event = 0, 
            display = 0x555557736720, extension = 35651594, evtype = 0, 
            cookie = 1314, data = 0x0}, pad = {140733193388036, 12816, 0, 
            93825027761952, 35651594, 1314, 0, 5126592, 1713691951399, 
            4518305596150, 4294967312, -4294967295, 0, 0, 257, 1097800351743, 
            0, 0, 0, 1065353216, 0, 1065353216, 1099511628031, 
            360644092100608}}
        this = 0x555557733950
#32 0x0000555555bc5992 in std::__invoke_impl<void, juce::XWindowSystem::initialiseXDisplay()::<lambda(int)>&, int>(std::__invoke_other, struct {...} &) (
    __f=...) at /usr/include/c++/10/bits/invoke.h:60
No locals.
#33 0x0000555555bc1f29 in std::__invoke_r<void, juce::XWindowSystem::initialiseXDisplay()::<lambda(int)>&, int>(struct {...} &) (__fn=...)
    at /usr/include/c++/10/bits/invoke.h:153
No locals.
#34 0x0000555555bbf169 in std::_Function_handler<void(int), juce::XWindowSystem::initialiseXDisplay()::<lambda(int)> >::_M_invoke(const std::_Any_data &, int &&--Type <RET> for more, q to quit, c to continue without paging--
) (__functor=..., __args#0=@0x7fffffffdf74: 5)
    at /usr/include/c++/10/bits/std_function.h:291
No locals.
#35 0x0000555555a208be in std::function<void (int)>::operator()(int) const (
    this=0x555557732ca0, __args#0=5)
    at /usr/include/c++/10/bits/std_function.h:622
No locals.
#36 0x0000555555a1bc31 in juce::InternalRunLoop::dispatchPendingEvents (
    this=0x555557732bf0)
    at ../../JUCE/modules/juce_events/native/juce_linux_Messaging.cpp:186
        insideFdReadCallback = {value = @0x555557732c48, originalValue = false}
        fdAndCallback = {first = 5, 
          second = {<std::_Maybe_unary_or_binary_function<void, int>> = {<std::unary_function<int, void>> = {<No data fields>}, <No data fields>}, <std::_Function_base> = {static _M_max_size = 16, static _M_max_align = 8, _M_functor = {
                _M_unused = {_M_object = 0x555557733950, 
                  _M_const_object = 0x555557733950, 
                  _M_function_pointer = 0x555557733950, 
                  _M_member_pointer = (void (std::_Undefined_class::*)(std::_Undefined_class * const)) 0x555557733950}, 
                _M_pod_data = "P9sWUU\000\000\000\000\000\000\000\000\000"}, 
              _M_manager = 0x555555bbf170 <std::_Function_handler<void(int), juce::XWindowSystem::initialiseXDisplay()::<lambda(int)> >::_M_manager(std::_Any_da--Type <RET> for more, q to quit, c to continue without paging--
ta &, const std::_Any_data &, std::_Manager_operation)>}, 
            _M_invoker = 0x555555bbf12c <std::_Function_handler<void(int), juce::XWindowSystem::initialiseXDisplay()::<lambda(int)> >::_M_invoke(const std::_Any_data &, int &&)>}}
        __for_range = std::vector of length 2, capacity 16 = {{first = 4, 
            second = {<std::_Maybe_unary_or_binary_function<void, int>> = {<std::unary_function<int, void>> = {<No data fields>}, <No data fields>}, <std::_Function_base> = {static _M_max_size = 16, static _M_max_align = 8, _M_functor = {
                  _M_unused = {_M_object = 0x555557732f00, 
                    _M_const_object = 0x555557732f00, 
                    _M_function_pointer = 0x555557732f00, 
                    _M_member_pointer = (void (std::_Undefined_class::*)(std::_Undefined_class * const)) 0x555557732f00, this adjustment 140737344885092}, 
                  _M_pod_data = "\000/sWUU\000\000d\321r\367\377\177\000"}, 
                _M_manager = 0x555555a249f2 <std::_Function_handler<void (int), juce::InternalMessageQueue::InternalMessageQueue()::{lambda(int)#1}>::_M_manager(std::_Any_data&, std::_Any_data const&, std::_Manager_operation)>}, 
              _M_invoker = 0x555555a249ae <std::_Function_handler<void (int), juce::InternalMessageQueue::InternalMessageQueue()::{lambda(int)#1}>::_M_invoke(std::_Any_data const&, int&&)>}}, {first = 5, 
            second = {<std::_Maybe_unary_or_binary_function<void, int>> = {<std::unary_function<int, void>> = {<No data fields>}, <No data fields>}, <std::_Function_base> = {static _M_max_size = 16, static _M_max_align = 8, _M_functor = {
--Type <RET> for more, q to quit, c to continue without paging--
                  _M_unused = {_M_object = 0x555557733950, 
                    _M_const_object = 0x555557733950, 
                    _M_function_pointer = 0x555557733950, 
                    _M_member_pointer = (void (std::_Undefined_class::*)(std::_Undefined_class * const)) 0x555557733950}, 
                  _M_pod_data = "P9sWUU\000\000\000\000\000\000\000\000\000"}, 
                _M_manager = 0x555555bbf170 <std::_Function_handler<void(int), juce::XWindowSystem::initialiseXDisplay()::<lambda(int)> >::_M_manager(std::_Any_data &, const std::_Any_data &, std::_Manager_operation)>}, 
              _M_invoker = 0x555555bbf12c <std::_Function_handler<void(int), juce::XWindowSystem::initialiseXDisplay()::<lambda(int)> >::_M_invoke(const std::_Any_data &, int &&)>}}}
        __for_begin = 
    {first = 5, second = {<std::_Maybe_unary_or_binary_function<void, int>> = {<std::unary_function<int, void>> = {<No data fields>}, <No data fields>}, <std::_Function_base> = {static _M_max_size = 16, static _M_max_align = 8, _M_functor = {_M_unused = {_M_object = 0x555557733950, _M_const_object = 0x555557733950, _M_function_pointer = 0x555557733950, _M_member_pointer = (void (std::_Undefined_class::*)(std::_Undefined_class * const)) 0x555557733950}, _M_pod_data = "P9sWUU\000\000\000\000\000\000\000\000\000"}, _M_manager = 0x555555bbf170 <std::_Function_handler<void(int), juce::XWindowSystem::initialiseXDisplay()::<lambda(int)> >::_M_manager(std::_Any_data &, const std::_Any_data &, std::_Manager_operation)>}, _M_invoker = 0x555555bbf12c <std::_Function_handler<void(int), juce::XWindowSy--Type <RET> for more, q to quit, c to continue without paging--
stem::initialiseXDisplay()::<lambda(int)> >::_M_invoke(const std::_Any_data &, int &&)>}}
        __for_end = 
    {first = 0, second = {<std::_Maybe_unary_or_binary_function<void, int>> = {<std::unary_function<int, void>> = {<No data fields>}, <No data fields>}, <std::_Function_base> = {static _M_max_size = 16, static _M_max_align = 8, _M_functor = {_M_unused = {_M_object = 0x0, _M_const_object = 0x0, _M_function_pointer = 0x0, _M_member_pointer = NULL}, _M_pod_data = '\000' <repeats 15 times>}, _M_manager = 0x0}, _M_invoker = 0x0}}
        fd = 5
        pfd = @0x55555743e918: {fd = 5, events = 1, revents = 0}
        __for_range = std::vector of length 2, capacity 2 = {{fd = 4, 
            events = 1, revents = 0}, {fd = 5, events = 1, revents = 0}}
        __for_begin = {fd = 5, events = 1, revents = 0}
        __for_end = {fd = 0, events = 0, revents = 0}
        sl = {lock_ = @0x555557732bf0}
        eventWasSent = false
#37 0x0000555555a1655f in juce::MessageManager::dispatchNextMessageOnSystemQueue (returnIfNoPendingMessages=false)
    at ../../JUCE/modules/juce_events/native/juce_linux_Messaging.cpp:300
        runLoop = 0x555557732bf0
#38 0x0000555555a10cdd in juce::MessageManager::runDispatchLoop (
    this=0x555557732b80)
--Type <RET> for more, q to quit, c to continue without paging--
    at ../../JUCE/modules/juce_events/messages/juce_MessageManager.cpp:128
No locals.
#39 0x0000555555a10179 in juce::JUCEApplicationBase::main ()
    at ../../JUCE/modules/juce_events/messages/juce_ApplicationBase.cpp:262
        libraryInitialiser = {<No data fields>}
        app = std::unique_ptr<juce::JUCEApplicationBase> = {
          get() = 0x555557732bb0}
#40 0x0000555555a1007e in juce::JUCEApplicationBase::main (argc=1, 
    argv=0x7fffffffe208)
    at ../../JUCE/modules/juce_events/messages/juce_ApplicationBase.cpp:240
No locals.
#41 0x00005555557043bc in main (argc=1, argv=0x7fffffffe208)
    at ../../JUCE/modules/juce_audio_plugin_client/juce_audio_plugin_client_Standalone.cpp:47
No locals.
(gdb) 
(gdb) 
(gdb) q
A debugging session is active.

	Inferior 1 [process 5432] will be killed.

Quit anyway? (y or n) y

**Comment by tomlem** [comment]
I have closed it unintentionally.

**Comment by tomlem** [comment]
I had closed it unintentionally.

**Comment by keinstein** [comment]
can you switch to the branch `lua5.2`, please?

Currently, the Ctrlr crashes because it does not know how to convert a non-ASCII string to Unicode (probably utf-8).

Which Editor did you use to generate that string: `"Loaded \273 180 bytes from sysex file"` ? Did you use the internal Lua editor or some kind of external Editor. Theoretically the source code should be UTF-8. The character \273 is neither UTF-8 nor ASCII.

So probably also my patch will fail.

Can you please send the panel it to me via email for further testing? Maybe we can avoid these crashes in debug mode.

My Address (will be deleted after some time): keinstein (strange symbol) users bullet sourceforge dot net

**Comment by tomlem** [comment]
Panel and .syx file sent via email.
Tomorrow I wil compile with branch lua 5.2

Thomas

**Comment by tomlem** [comment]
Immediately after compilation (yesterday) the program correctly loaded the .syx file into the ctrlr memory block and then sent it correctly to the sound module as well. That's why I said in my first comment that the MIDI  in and out went correctly. Only on subsequent attempts did the program freeze. There may be something stuck in memory or in the configuration file.

**Comment by tomlem** [comment]
I've got these errors while trying to compile ctrlr-lua5.2.
On my pc both Lua 5.1.5 and Lua 5.3.3 are installed
```
~/build2/ctrlr-lua5.2/Builds/LinuxMakefile$ make -f pchbuild.mk CONFIG=Debug Standalone
Package lua5.1 was not found in the pkg-config search path.
Perhaps you should add the directory containing `lua5.1.pc'
to the PKG_CONFIG_PATH environment variable
Package 'lua5.1', required by 'lua5.2-bitop', not found
Makefile:295: *** target pattern contains no '%'. Stop.
```

**Comment by keinstein** [comment]
`apt-get install liblua5.2-dev`

**Comment by tomlem** [comment]
sudo apt-get install liblua5.2-dev
[sudo] wachtwoord voor tom: 
Pakketlijsten worden ingelezen... Klaar
Boom van vereisten wordt opgebouwd... Klaar
De statusinformatie wordt gelezen... Klaar 
liblua5.2-dev is reeds de nieuwste versie (5.2.4-1.1+b3).
0 opgewaardeerd, 0 nieuw geïnstalleerd, 0 te verwijderen en 0 niet opgewaardeerd

**Comment by tomlem** [comment]
After installing liblua5.1-dev and after replacing lines 51 and 93 of the Makefile with
`VST3_PLATFORM_ARCH := $(shell uname -m)`
it is compiling now.

**Comment by tomlem** [comment]
Back trace of GDB for the following actions:

Opening panel (the one I sent you), setting up Midi channels and opening Tools/MIDI window -- suceeded;
loading the same .syx file as I sent you  -- succeeded;
sending this .syx file to the sound module -- succeeded;
receiving a bulk dump of the setup from these sound module -- succeeded;
sending the received setup bulk dump to a file on the pc -- succeeded;
opening preferences and setting some preferences -- succeeded;
closing this panel -- succeeded;
opening a new panel -- while changing directory on pc in search for other panel -- freezed

See attached file.

[gdb_backtrace_2021_06_22.txt](https://github.com/RomanKubiak/ctrlr/files/6695069/gdb_backtrace_2021_06_22.txt)

Thomas

**Comment by keinstein** [comment]
Can you reproduce this crash?
I assume as soon as you install `kdialog` or `zenity` the problem disappears.

Some explanation: JUCE tries to display the text “Go up to parent directory” as tooltip and assumes that the file chooser dialog has already a tooltip open. JUCE consideres this a bug (in JUCE code).

This is one of the many Linux bugs that JUCE has.

**Comment by tomlem** [comment]
After installing kdialog I couldn't reproduce the previous crash. I loaded another, a work in progress panel and when trying to display some lua structures in the lua window the program was freezing.

[gdb_backtrace_2021_06_22_20:15.txt](https://github.com/RomanKubiak/ctrlr/files/6696539/gdb_backtrace_2021_06_22_20.15.txt)

**Comment by keinstein** [comment]
@RomanKubiak can you have a look at `Source/Core/CtrlrPanel/CtrlrPanelMIDISnapshot.{cpp,h}`?

It seems that there are several ways to stop the thread in this class. The Destructor unconditionally stops the thread, which hits a JUCE assertion if the thread is not running.

**Comment by tomlem** [comment]
_I'm testing the lua2 version some more._

Some lua functions, especially those using the bitop library, are no longer working. I added `local bit = require("bit")` as first line in these functions (b.i. the functions lsTN1, lsTN2... in the panel I sent you).
**Error message:**
```
Callback error: [lsTN1] 
At line [-1]: [C]
What: C
Namewhat: method
Name: sendMidiMessageNow
Error message: No matching overload found, candidates:
void sendMidiMessageNow(CtrlrPanel&,CtrlrMidiMessage&) 
At line [-1]: [C]
What: C
Namewhat: method
Name: sendMidiMessageNow
Error message: No matching overload found, candidates:
void sendMidiMessageNow(CtrlrPanel&,CtrlrMidiMessage&).
Method disabled
JUCE Assertion failure in juce_DragAndDropContainer.cpp:585
```
[gdb_backtrace_NoMatchingOverloadFound_2021_06_25_11:11.txt](https://github.com/RomanKubiak/ctrlr/files/6715084/gdb_backtrace_NoMatchingOverloadFound_2021_06_25_11.11.txt)

In the lua window is it no longer possible to close a lua function by clicking along its name in the tab bar.

**Comment by keinstein** [comment]
> @RomanKubiak can you have a look at `Source/Core/CtrlrPanel/CtrlrPanelMIDISnapshot.{cpp,h}`?

I added a suggestion to fix this issue to branch `keinsteins_fixes`



> I added `local bit = require("bit")` as first line in these functions (b.i. the functions lsTN1, lsTN2... in the panel I sent you).

I didn't recognise that it doesn't work. Ctrlr has two functions that initialise Lua and probably I updated the wrong one. Both should load `bit` by default, now.



> JUCE Assertion failure in juce_DragAndDropContainer.cpp:585

This is not related to sendMidiMessageNow. It is because Ctrlr left it to JUCE to guess the pointer that generates the drag event. Now, I changed the code that the pointer device from the event is sent to JUCE so that it can't guess wrongly.

The sendMidiMessageNow issue is something different. How do other versions behave? Whats on Windows, Mac/iOS other Linux builds?

**Comment by keinstein** [comment]
I tried your two panels, but I didn't get the Lua error. If it persists, please make a minimal example panel, which has just one button
that triggers the error, so that I can debug it.

**Comment by tomlem** [comment]
As you suggested I've made a simple test panel.
Its is working using Ctrlr 5.3.201 (of course without ``local` bit = require("bit")`)  but not using lua 5.2 (with `local bit = require("bit")` as first argument in the function.

[gdb_backtrace_Bitoptest_2021_06_26_17:00.txt](https://github.com/RomanKubiak/ctrlr/files/6720290/gdb_backtrace_Bitoptest_2021_06_26_17.00.txt)
[Ctrlr Panel_1_0_Hell-O-Kitty_BitopTest_2021-06-26_13-04.zip](https://github.com/RomanKubiak/ctrlr/files/6720291/Ctrlr.Panel_1_0_Hell-O-Kitty_BitopTest_2021-06-26_13-04.zip)

**Comment by keinstein** [comment]
CtrlrMidiMessage has no constructor that takes a Lua array and automatic conversion to and from CtrlrLuaObjectWrapper is not set up. As I can't find any information about automatic type conversion in Lua I assume that code like `CrtlrMidiMessage({0xf0, ... , 0xf7})` should not work with any Lua version.

@RomanKubiak What is the right way to do?

**Comment by keinstein** [comment]
@tomlem does https://github.com/RomanKubiak/ctrlr/pull/400 work for you?

**Comment by tomlem** [comment]
Hi Tobias,

That's really good news.

At the moment I don't have access to my PC.
I will definitely test it out and report on it in a few days.

Thomas

**Comment by tomlem** [comment]
Hi Tobias,

After executing 
`git clone https://github.com/keinstein/ctrlr.git` 
and after reading the updated documentation (ctrlr/README.md) to know what dependency packages I had to install (on my system the specific version 1.0 of libusb was required),
I executed
`cd ctrlr/Builds/LinuxMakefile/`
and then
 `make -f pchbuild.mk CONFIG=Debug Standalone`

The compilation process went well.

After playing with this version for an hour or so, my impression of it is very good.
In the next days I will test it even more in depth.

Thomas

**Comment by tomlem** [comment]
Some remarks @keinstein 

-  `LANG=C ./build.sh Release Standalone` also compiles without errors
- When I try to close a panel (File>Close) there appears a requester (Save Discard Cancel). When clicking on Save or on Discard not only the panel is closing but Ctrlr also.
- The lua method "lsTN1" in the attached panel gives this error in the terminal (I started Ctrlr from terminal): 
`Callback error: [lsTN1]` 
`At line [-1]: [C]`
`Error message: No matching overload found, candidates:`
`void __init(luabind::argument const&,JMemoryBlock&)`
`void __init(luabind::argument const&,MidiMessage const&)`
`void __init(luabind::argument const&,CtrlrLuaObjectWrapper const&)`
`void __init(luabind::argument const&,String const&)`
`Method disabled`

[Ctrlr Panel_1_0_Hell-O-Kitty_BitopTest_2021-06-26_13-04.zip](https://github.com/RomanKubiak/ctrlr/files/7830252/Ctrlr.Panel_1_0_Hell-O-Kitty_BitopTest_2021-06-26_13-04.zip)

**Comment by keinstein** [comment]
Regarding the issues: I have no idea. Can you confirm that it works with your version on Linux?

**Comment by keinstein** [comment]
I was too fast. Your second point is unrelated to this issue. It is merely a side effect that Ctrlr does not crash anymore when you rename a function. The code checks that the name of the function is defined after compiling the source. Otherwise it gives the warning. At that point I didn't think about the fact that a table can be called as a function under certain circumstances.

That code is just diagnostic. The function cannot be called from the panel if it is not defined.

Regarding the Save/Discard etc. Its probably completely unrelated to this patches. These are bugs that have been available for a long time. I compile using the address sanitizer, which is very picky about such problems. Actually it has be developed to find these issues. If you remove -fsanitize=address from the compiler options, Ctrlr may and may not crash at this point. If it does not crash you have corrupted memory. This may not be a problem but it is still a bug. I'd suggest to use the address sanitizer always on any platform that supports it (Mac OS X or Linux or MingW*) in order to early find such issues, not only after someone fixed another one.

**Comment by tomlem** [comment]
I Can you confirm that it is working really good on my system
(`System: Kernel: 5.10.0-10-amd64 x86_64 bits: 64 Desktop: GNOME 3.38.6 Distro: Debian GNU/Linux 11 (bullseye)`)
EXCEPT the problem mentioned in the third remark in my previous post.

**Comment by keinstein** [comment]
Could we move the discussion of the build to https://github.com/RomanKubiak/ctrlr/pull/400 , please?

**Comment by keinstein** [comment]
Just for the records: Automatic conversion from Lua objects to to CtrlLuaObjectWrapper is not registered with luabind. And then: what is the best? JMemoryBlock, MidiMessage, String? I try to implement the automatic convertion to CtrlLuaObjectWrapper. And I expect it to break lots of things as it is theoretically used in many places but practically in none of them.

**Comment by keinstein** [comment]
@tomlem Take a look at https://github.com/keinstein/ctrlr/tree/master, please
